collaboration-service/index.js

```diff
@@ -0,0 +1,103 @@
+import express from "express";
+import cors from "cors";
+import { createServer } from "http";
+import mongoose from "mongoose";
+import "dotenv/config";
+import { Server, Socket } from "socket.io";
+import Room from "./model/room-model.js";
+import collaborationRoutes from "./routes/collaboration-routes.js";
+
+const uri =
+  process.env.ENV == "PROD"
+    ? process.env.DB_CLOUD_URI
+    : process.env.DB_LOCAL_URI;
+
+mongoose
+  .connect(uri)
+  .then((x) =>
+    console.log(
+      `Connected to MongoDB! Database name: "${x.connections[0].name}"`
+    )
+  )
+  .catch((err) => console.error("Error connecting to MongoDB", err.reason));
+
+const app = express();
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
+app.use(cors());
+app.options("*", cors());
+
+// Routes tested via Postman
+app.use("/collab", collaborationRoutes);
+
+const httpServer = createServer(app);
+const PORT = process.env.PORT || 8003;
+
+httpServer.listen(PORT, () => {
+  console.log(`Listening on port ${PORT}`);
+});
+
+// Sockets tested using https://socketio.bloggernepal.com/4.0.1/
+// Also tested via Postman
+const io = new Server(httpServer, {
+  cors: {
+    origin: true,
+  },
```

Comment:
Can clarify why `origin: true` instead of say, `origin: http://localhost:3000` or `origin: '*'`?

Reply:
Online guide mentioned this as a way to address certain errors in socket connection. Will take a further look into it.

Reply:
Can send the link? I can take a look.

https://socket.io/docs/v4/handling-cors/ 
https://stackoverflow.com/questions/24058157/socket-io-node-js-cross-origin-request-blocked

Because I thought it would be `origin: *`, meaning that the origin of clients can be anywhere. If let's say it is `origin: localhost:3000`, then only react app on port 3000 can access the resource. Another microservice on port 8080 (example) won't be able to access this collab service.

Reply:
Went back to check and found that I mistook it for another tutorial I used. Yep will follow using localhost:3000.